File: backend/metier_generator/templates/threed/tests_centralises/plugins/test_plugins.py
"""
Tests centralisés pour les plugins threed (Python)
"""

from ...plugins import load_plugin, validate_plugin


def test_auth_plugin():
    plugin = load_plugin("auth")
    assert plugin is not None
    assert validate_plugin(plugin) is True


def test_invalid_plugin():
    invalid = {"name": "Fake", "hooks": []}
    assert validate_plugin(invalid) is False

File: backend/metier_generator/templates/threed/tests_centralises/plugins/test_plugins.js
// Tests avancés pour les plugins threed
const { loadPlugin, validatePlugin } = require('../../api/plugins');
describe('Plugins threed', () => {
  it('doit charger le plugin d’authentification et le valider', () => {
    const plugin = loadPlugin('auth');
    expect(plugin).toBeDefined();
    expect(validatePlugin(plugin)).toBe(true);
  });
  it('doit refuser un plugin non conforme', () => {
    const invalid = { name: 'Fake', hooks: [] };
    expect(validatePlugin(invalid)).toBe(false);
  });
});
